### PR TITLE
Introduce longterm environment to TSI resource hierarchy

### PR DESCRIPTION
**Pull request by sandshadow**
Add a new L1 Sku for long-term environments. Introduces a new "kind" property on Environment that is used as a discriminator to determine whether the environment is "Standard" or "LongTerm". This is basically the same pattern as we use when creating event sources. 

Changes are on a new preview api version. This api is initially hidden behind a feature flag for private preview of long-term environments. 

Review by ARM team is being added on the pull-request into the TimeSeriesInsights fork so we can get signoff early rather than wait until it goes to master.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by sandshadow** [comment]
@ravbhatnagar 
Can you review this change to the TSI swagger spec? It's a PR into the team's fork of the repository so we can get the ARM team's review before we are ready to expose publicly.

File: specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/preview/2018-08-15-preview/timeseriesinsights.json [hunk 1]
```diff
@@ -1201,12 +1202,56 @@
       ],
       "description": "The sku determines the capacity of the environment, the SLA (in queries-per-minute and total capacity), and the billing rate."
```
Comment (yeskarthik) [inline specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/preview/2018-08-15-preview/timeseriesinsights.json]: Also determines the "storage type" now?

Comment (sandshadow) [reply specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/preview/2018-08-15-preview/timeseriesinsights.json]: I updated the description to indicate the Sku determines the type of environment.

**Comment by sandshadow** [comment]
Got email signoff from @ravbhatnagar

**Pull request merged**